### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by DarthPigrum**

### CODE REVIEW

**Comment by tshemsedinov** [comment]
Change PR and commit title to `Improve error handling` @DarthPigrum

**Comment by DarthPigrum** [comment]
Done(not sure the best way possible on git)

**Comment by tshemsedinov** [comment]
Please don't use revert commits, if you need to drop previous you can use `git reset --hard HEAD~1` or if you want to drop two previous commits `git rebase -i HEAD~3` then mark needed commits to delete and `git push -f`. If something goes wrong you can ask for help but don't close this PR and open new for the same task. @DarthPigrum

**Comment by DarthPigrum** [comment]
Done

File: metalog.js [hunk 1]
```diff
@@ -121,8 +121,9 @@ Logger.prototype.close = function() {
       const fileName = this.file;
       this.emit('close');
       fs.stat(fileName, (err, stats) => {
-        if (err || stats.size > 0) return;
-        fs.unlink(this.file, () => {});
+        if (err) { console.log(err); return; }
```
Comment (tshemsedinov) [inline metalog.js]:
```suggestion
        if (err) {
          console.log(err);
          return;
        }
```

File: metalog.js [hunk 2]
```diff
@@ -121,8 +121,9 @@ Logger.prototype.close = function() {
...
+        if (stats.size > 0) return;
+        fs.unlink(this.file, (err) => { console.log(err); });
```
Comment (tshemsedinov) [inline metalog.js]:
```suggestion
        fs.unlink(this.file, (err) => {
          console.log(err);
        });
```

File: metalog.js [hunk 3]
```diff
@@ -131,7 +132,7 @@ Logger.prototype.close = function() {
 Logger.prototype.rotate = function() {
   if (!this.keepDays) return;
   fs.readdir(this.path, (err, files) => {
-    if (err) return;
+    if (err) { console.log(err); return; }
```
Comment (tshemsedinov) [inline metalog.js]:
```suggestion
    if (err) {
      console.log(err);
      return;
    }
```

File: metalog.js [hunk 4]
```diff
@@ -143,7 +144,7 @@ Logger.prototype.rotate = function() {
       fileTime = new Date(fileName.substring(0, 10)).getTime();
       fileAge = Math.floor((time - fileTime) / DAY_MILLISECONDS);
       if (fileAge > 1 && fileAge > this.keepDays - 1) {
-        fs.unlink(this.path + '/' + fileName, common.emptiness);
+        fs.unlink(this.path + '/' + fileName, (err) => { console.log(err); });
```
Comment (tshemsedinov) [inline metalog.js]:
```suggestion
        fs.unlink(this.path + '/' + fileName, (err) => {
          console.log(err);
        });
```

**Comment by tshemsedinov** [comment]
@DarthPigrum you can add new commits to this PR or even `git commit fixup=<hash>`

**Comment by tshemsedinov** [comment]
Landed in f55dbf5

**Comment by tshemsedinov** [comment]
Thanks for first contribution @DarthPigrum, your changes are squashed with my fixes and landed to master branch.

**Pull request closed without merging**